internal/util/function/bedrock_embedding_provider.go

```diff
@@ -0,0 +1,209 @@
+	}
+	if region == "" {
+		return nil, fmt.Errorf("Missing region. Please pass `region` param.")
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(
+			awsAccessKeyId, awsSecretAccessKey, "")),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return bedrockruntime.NewFromConfig(cfg), nil
+}
+
+func NewBedrockEmbeddingProvider(fieldSchema *schemapb.FieldSchema, functionSchema *schemapb.FunctionSchema, c BedrockClient) (*BedrockEmbeddingProvider, error) {
+	fieldDim, err := typeutil.GetDim(fieldSchema)
+	if err != nil {
+		return nil, err
+	}
+	var awsAccessKeyId, awsSecretAccessKey, region, modelName string
+	var dim int64
+	normalize := true
+
+	for _, param := range functionSchema.Params {
+		switch strings.ToLower(param.Key) {
+		case modelNameParamKey:
+			modelName = param.Value
+		case dimParamKey:
+			dim, err = strconv.ParseInt(param.Value, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("dim [%s] is not int", param.Value)
+			}
+
+			if dim != 0 && dim != fieldDim {
+				return nil, fmt.Errorf("Field %s's dim is [%d], but embeding's dim is [%d]", functionSchema.Name, fieldDim, dim)
+			}
+		case awsAccessKeyIdParamKey:
+			awsAccessKeyId = param.Value
+		case awsSecretAccessKeyParamKey:
+			awsSecretAccessKey = param.Value
+		case regionParamKey:
+			region = param.Value
+		case normalizeParamKey:
+			switch strings.ToLower(param.Value) {
+			case "false":
+				normalize = false
+			default:
+				return nil, fmt.Errorf("Illegal [%s:%s] param, ", normalizeParamKey, param.Value)
+			}
+		default:
+		}
+	}
+
+	if modelName != BedRockTitanTextEmbeddingsV2 {
+		return nil, fmt.Errorf("Unsupported model: %s, only support [%s]",
+			modelName, BedRockTitanTextEmbeddingsV2)
+	}
+	var client BedrockClient
+	if c == nil {
+		client, err = createBedRockEmbeddingClient(awsAccessKeyId, awsSecretAccessKey, region)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		client = c
+	}
+
+	return &BedrockEmbeddingProvider{
+		client:        client,
+		fieldDim:      fieldDim,
+		modelName:     modelName,
+		embedDimParam: dim,
+		normalize:     normalize,
+		maxBatch:      1,
+		timeoutSec:    30,
+	}, nil
+}
+
+func (provider *BedrockEmbeddingProvider) MaxBatch() int {
+	return 12 * provider.maxBatch
+}
+
+func (provider *BedrockEmbeddingProvider) FieldDim() int64 {
+	return provider.fieldDim
+}
+
+func (provider *BedrockEmbeddingProvider) CallEmbedding(texts []string, batchLimit bool, _ string) ([][]float32, error) {
+	numRows := len(texts)
+	if batchLimit && numRows > provider.MaxBatch() {
+		return nil, fmt.Errorf("Bedrock text embedding supports up to [%d] pieces of data at a time, got [%d]", provider.MaxBatch(), numRows)
+	}
+
+	data := make([][]float32, 0, numRows)
+	for i := 0; i < numRows; i += 1 {
```

Comment:
I assume batch embedding not supported in bedrock? would be best to add a comment